src/offers/client_impls.rs

```diff
@@ -135,6 +137,19 @@ impl InvoicePayer for Client {
         let query_req = tonic_lnd::lnrpc::QueryRoutesRequest {
             amt_msat: msats as i64,
             blinded_payment_paths: vec![blinded_payment_paths],
+            fee_limit: Some(tonic_lnd::lnrpc::FeeLimit {
+                limit: if let Some(fee_limit) = fee_limit {
+                    Some(tonic_lnd::lnrpc::fee_limit::Limit::FixedMsat(
+                        fee_limit as i64,
```

Comment:
This seems odd. Here could be an overflow for big values because LND is expecting int64 values.
So double checkin what LND does in `marshall_utils.go:31` they, then force it to uint to calculate the fixed rate. So in that case we would have an issue.
So I'd suggest either set the same fee_limit or validate in server and CLI that we won't go over it

Reply:
maybe uint32?